internal/otelcollector/config/metric/agent/receivers_test.go

```diff
@@ -55,6 +55,169 @@ func TestReceivers(t *testing.T) {
+		}, BuildOptions{})
+
+		require.Nil(t, collectorConfig.Receivers.PrometheusAppPods)
+		require.Nil(t, collectorConfig.Receivers.PrometheusIstio)
+
+		expectedKubeletStatsReceiver := KubeletStatsReceiver{
+			CollectionInterval: "30s",
+			AuthType:           "serviceAccount",
+			Endpoint:           "https://${MY_NODE_NAME}:10250",
+			InsecureSkipVerify: true,
+			MetricGroups:       []MetricGroupType{"node"},
+			Metrics: KubeletStatsMetricsConfig{
+				ContainerCPUUsage:            MetricConfig{Enabled: true},
+				ContainerCPUUtilization:      MetricConfig{Enabled: false},
+				K8sPodCPUUsage:               MetricConfig{Enabled: true},
+				K8sPodCPUUtilization:         MetricConfig{Enabled: false},
+				K8sNodeCPUUsage:              MetricConfig{Enabled: true},
+				K8sNodeCPUUtilization:        MetricConfig{Enabled: false},
+				K8sNodeCPUTime:               MetricConfig{Enabled: false},
+				K8sNodeMemoryMajorPageFaults: MetricConfig{Enabled: false},
+				K8sNodeMemoryPageFaults:      MetricConfig{Enabled: false},
+				K8sNodeNetworkIO:             MetricConfig{Enabled: false},
+				K8sNodeNetworkErrors:         MetricConfig{Enabled: false},
+			},
+			ExtraMetadataLabels: []string{
+				"k8s.volume.type",
+			},
+		}
+		require.Equal(t, expectedKubeletStatsReceiver, *collectorConfig.Receivers.KubeletStats)
+	})
+
+	t.Run("runtime input enabled with only volume metrics enabled", func(t *testing.T) {
+		collectorConfig := sut.Build([]telemetryv1alpha1.MetricPipeline{
+			testutils.NewMetricPipelineBuilder().
+				WithRuntimeInput(true).
+				WithRuntimeInputContainerMetrics(false).
+				WithRuntimeInputPodMetrics(false).
+				WithRuntimeInputNodeMetrics(false).
+				WithRuntimeInputVolumeMetrics(true).
+				Build(),
+		}, BuildOptions{})
+
+		require.Nil(t, collectorConfig.Receivers.PrometheusAppPods)
+		require.Nil(t, collectorConfig.Receivers.PrometheusIstio)
+
+		expectedKubeletStatsReceiver := KubeletStatsReceiver{
+			CollectionInterval: "30s",
+			AuthType:           "serviceAccount",
+			Endpoint:           "https://${MY_NODE_NAME}:10250",
+			InsecureSkipVerify: true,
+			MetricGroups:       []MetricGroupType{"volume"},
+			Metrics: KubeletStatsMetricsConfig{
+				ContainerCPUUsage:            MetricConfig{Enabled: true},
+				ContainerCPUUtilization:      MetricConfig{Enabled: false},
+				K8sPodCPUUsage:               MetricConfig{Enabled: true},
+				K8sPodCPUUtilization:         MetricConfig{Enabled: false},
+				K8sNodeCPUUsage:              MetricConfig{Enabled: true},
+				K8sNodeCPUUtilization:        MetricConfig{Enabled: false},
+				K8sNodeCPUTime:               MetricConfig{Enabled: false},
+				K8sNodeMemoryMajorPageFaults: MetricConfig{Enabled: false},
+				K8sNodeMemoryPageFaults:      MetricConfig{Enabled: false},
+				K8sNodeNetworkIO:             MetricConfig{Enabled: false},
+				K8sNodeNetworkErrors:         MetricConfig{Enabled: false},
+			},
+			ExtraMetadataLabels: []string{
+				"k8s.volume.type",
+			},
```

Comment:
This would be a good use for a table test as the difference in input and expected output for each of those tests is quite hard to spot